src/jsone_encode.erl

```diff
@@ -306,6 +307,8 @@ array_values([],       Nexts, Buf, Opt) -> next(Nexts, <<(pp_newline(Buf, Nexts,
 array_values([X | Xs], Nexts, Buf, Opt) -> value(X, [{array_values, Xs} | Nexts], Buf, Opt).
 
 -spec object(jsone:json_object_members(), [next()], binary(), opt()) -> encode_result().
+object(Members, Nexts, Buf, ?OPT{canonical_form = true}=Opt) ->
+  object_members(lists:keysort(1, Members), Nexts, pp_newline(<<Buf/binary, ${>>, Nexts, 1, Opt), Opt);
```

Comment:
@pichi to have the sort done on the key only and not on term of the list which is a tuple and then would make it sorted differently depending on the platform.